src/include/utils/tiledb_helpers.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file   tiledb_helpers.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Helper functions for certain TileDB operations.
+ *
+ */
+
+#ifndef TILEDB_HELPERS_H
+#define TILEDB_HELPERS_H
+
+#include <tiledb/tiledb>
+
+namespace tiledb_helpers {
+
+namespace detail {
+class StatsCollectionScope final {
+public:
+  explicit StatsCollectionScope(const std::string &name) {
+    tiledb::Stats::enable();
+    tiledb::Stats::reset();
+    name_ = name;
+  }
+
+  ~StatsCollectionScope() {
+    fprintf(stderr, "---------- STATS %s ------------\n", name_.c_str());
+    tiledb::Stats::dump(stderr);
```

Comment:
Where the output goes should be configurable using an `ostream` with a default to `stdout`.  (I recommend using `stdout` as the default `ostream` as it is easier to grep for things -- separately filtering `stderr` from `stdout` is really annoying.)

Again, by using a singleton dictionary, with deferred display of the stats messages, it is easy to choose where to send the output.  The processing of messages is done through the dictionary, not the stats collector.

Reply:
Done. The file stats are written to is configurable with a `stats_file` variable. For the Python API it defaults to stdout (and cannot change).